spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -112,10 +115,18 @@
 
 	private volatile boolean running = false;
 
-	private Executor taskExecutor;
+	private Executor consumerTaskExecutor;
+
+	private AsyncListenableTaskExecutor listenerTaskExecutor;
 
 	private ErrorHandler errorHandler = new LoggingErrorHandler();
 
+	private long pauseAfter = DEFAULT_PAUSE_AFTER;
```

Comment:
I am wondering if we shouldn't calculate this based on the `heartbeat.interval.ms` property of the consumer (possibly lower than that), otherwise if the value is set too high (e.g. the 10s interval set by default is higher than the default `heartbeat.interval.ms` of 3s) this will not be effective.